x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/physical/local/PushStatsToSource.java

```diff
@@ -94,15 +94,19 @@ private Tuple<List<Attribute>, List<EsStatsQueryExec.Stat>> pushableStats(
                             // check if regular field
                             else {
                                 if (target instanceof FieldAttribute fa) {
-                                    var fName = fa.name();
+                                    var fName = fa.fieldName();
                                     if (context.searchStats().isSingleValue(fName)) {
-                                        fieldName = fa.name();
+                                        fieldName = fName;
                                         query = QueryBuilders.existsQuery(fieldName);
                                     }
                                 }
                             }
                             if (fieldName != null) {
                                 if (count.hasFilter()) {
+                                    // Note: currently, it seems like we never actually perform stats pushdown if we reach here.
```

Comment:
It seems the question of what exactly gets pushed down and why is subtle. I wonder if we want some documentation on this?

Reply:
> // Note: currently, it seems like we never actually perform stats pushdown if we reach here.

We evolved to this. Before the aggs filter were extracted into an upstream filter and pushed down, this code was "alive".

Reply:
Yep yep, and if we manage to evolve the optimization to multiple stats, this code could come alive again.

There's an argument to be made that it should be deleted as long as it's dead - but properly double checking that this code path _really is never ever used atm_ was beyond the scope that I could allocate to this bug fix.

So leaving a comment was the next best thing for the time being, I think  :)